### DISCUSSION TITLE
How do you close a database connection - can not find any examples or docs

### DISCUSSION THREAD

**Discussion by pkasson** [duckdb/duckdb#14357] [Q&A]
How do you close a database connection - can not find any examples or docs

**Comment by soerenwolfers** [comment]
Using which client API? E.g. https://duckdb.org/docs/api/python/reference/#duckdb.DuckDBPyConnection.close

**Reply by pkasson** [reply]
Swift - SPM 0.10.3

**Reply by soerenwolfers** [reply]
doesn't seem like there is an explicit close method, only a `deinit`. You should report it at the swift repo https://github.com/duckdb/duckdb-swift/blob/main/Sources/DuckDB/Connection.swift

**Reply by pkasson** [reply]
So a deinit method would safely close the database ?

**Reply by soerenwolfers** [reply]
it does, but only at deallocation time. I reckon there are situations, you might wanna close it earlier.

**Comment by pkasson** [comment]
For non memory capabilities, do you simple do nothing, or is there a close method

**Comment by pkasson** [comment]
Ok ... so if you do nothing, then that is the same as doing nothing to close a databases.
